tests/test_AggregateGroupData.cpp

```diff
@@ -0,0 +1,589 @@
+/*
+  Copyright 2018 Statoil ASA
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#define BOOST_TEST_MODULE Aggregate_Group_Data
+#include <opm/output/eclipse/AggregateGroupData.hpp>
+
+#include <boost/test/unit_test.hpp>
+
+#include <opm/output/eclipse/AggregateWellData.hpp>
+
+#include <opm/output/eclipse/VectorItems/intehead.hpp>
+#include <opm/output/eclipse/VectorItems/well.hpp>
+
+#include <opm/output/data/Wells.hpp>
+
+#include <opm/parser/eclipse/Deck/Deck.hpp>
+#include <opm/parser/eclipse/Parser/Parser.hpp>
+#include <opm/parser/eclipse/EclipseState/EclipseState.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Schedule.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/SummaryState.hpp>
+
+#include <exception>
+#include <stdexcept>
+#include <utility>
+#include <vector>
+#include <iostream>
+#include <cstddef>
+
+struct MockIH
+{
+    MockIH(const int numWells,
+	   
+	   const int igrpPerGrp	 = 101,  // no of data elements per group in IGRP array
+	   const int sgrpPerGrp  = 112,  // number of data elements per group in SGRP array
+	   const int xgrpPerGrp  = 180,  // number of data elements per group in XGRP array
+	   const int zgrpPerGrp  =   5);  // number of data elements per group in XGRP array
+    
+
+    std::vector<int> value;
+
+    using Sz = std::vector<int>::size_type;
+
+    Sz nwells;
+    Sz nwgmax;
+    Sz ngmaxz;
+    Sz nigrpz;
+    Sz nsgrpz;
+    Sz nxgrpz;
+    Sz nzgrpz;
+};
+
+MockIH::MockIH(const int numWells,
+	       const int igrpPerGrp,
+	       const int sgrpPerGrp,
+	       const int xgrpPerGrp,
+	       const int zgrpPerGrp)
+    : value(411, 0)
+{
+    using Ix = ::Opm::RestartIO::Helpers::VectorItems::intehead;
+
+    this->nwells = this->value[Ix::NWELLS] = numWells;
+    
+    this->ngmaxz = this->value[Ix::NGMAXZ] = 5;
+    this->nwgmax = this->value[Ix::NWGMAX] = 4;
+    this->nigrpz = this->value[Ix::NIGRPZ] = igrpPerGrp;
+    this->nsgrpz = this->value[Ix::NSGRPZ] = sgrpPerGrp;
+    this->nxgrpz = this->value[Ix::NXGRPZ] = xgrpPerGrp;
+    this->nzgrpz = this->value[Ix::NZGRPZ] = zgrpPerGrp;
+}
+
+namespace {
+    Opm::Deck first_sim()
+    {
+        // Mostly copy of tests/FIRST_SIM.DATA
+        const auto input = std::string {
```

Comment:
The type must be *somewhere* - why nut just:
```
const std::string input = R"( ... )";
```

Reply:
OK - Changed.